pkg/typesystem/typesystem.go

```diff
@@ -126,23 +126,45 @@ func Difference(base *openfgapb.Userset, sub *openfgapb.Userset) *openfgapb.User
 }
 
 type TypeSystem struct {
+	// [objectType] => typeDefinition
 	typeDefinitions map[string]*openfgapb.TypeDefinition
-	modelID         string
-	schemaVersion   string
+	// [objectType] => [relationName] => relation
+	relations     map[string]map[string]*openfgapb.Relation
```

Comment:
What I'd really hope to see in the near term is the `openfgapb.TypeDefinition` to change altogether. The protobuf definition for a TypeDefinition should be:

```
message TypeDefintion {
    string type = 1;
    map<string, Relation> relations = 2;
}
```

The current definition involving `map<string, Userset>` requires this extra construction AND it also necessitates the `metadata` as well.

Reply:
Left an issue in the `openfga/api` repository to address this.

https://github.com/openfga/api/issues/77